### PR TITLE
- Create the common transport and threadpool in the main loop.

### PR DESCRIPTION
**Pull request by baldersheim**
- Also use the common transport for config subscriptions.
- Put The TransportConfig in the fnet namespace.

@havardpe PR
@geirst @vekterli FYI

### CODE REVIEW

File: searchcore/src/apps/proton/proton.cpp [hunk 1]
```diff
@@ -186,8 +190,14 @@ App::Main()
         LOG(debug, "serviceidentity: '%s'", params.serviceidentity.c_str());
         LOG(debug, "subscribeTimeout: '%" PRIu64 "'", params.subscribeTimeout);
         std::chrono::milliseconds subscribeTimeout(params.subscribeTimeout);
-        config::ConfigUri identityUri(params.identity);
-        protonUP = std::make_unique<proton::Proton>(identityUri, _argc > 0 ? _argv[0] : "proton", subscribeTimeout);
+        FastOS_ThreadPool threadPool(128_Ki);
+        uint32_t numProcs = std::thread::hardware_concurrency();
+        FNET_Transport transport(fnet::TransportConfig(std::max(1u, std::min(4u, numProcs/8))));
```
Comment (geirst) [inline searchcore/src/apps/proton/proton.cpp:195]: Consider adding a helper function for calculating TransportConfig.

File: searchcore/src/apps/proton/proton.cpp [hunk 2]
```diff
@@ -228,6 +238,7 @@ App::Main()
                 EV_STOPPING("servicelayer", "clean shutdown");
             }
             protonUP.reset();
+            transport.ShutDown(true);
```
Comment (havardpe) [inline searchcore/src/apps/proton/proton.cpp:241]: consider shutting down transport before deleting proton, to reduce chance of core dumps

**Review by geirst — COMMENTED** [review]
The changes in proton looks good. @havardpe should look at the other parts.

**Review by havardpe — COMMENTED** [review]
deleting stuff using the shared transport before shutting down the transport may lead to unfortunate events.

File: searchcore/src/apps/proton/proton.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/apps/proton/proton.cpp]: Xixed

**Pull request merged**